embedded-graphics/src/primitives/sector.rs

```diff
@@ -0,0 +1,798 @@
+        let outer_threshold = circle::diameter_to_threshold(stroke_area.diameter);
+
+        let line_a = stroke_area.line_from_angle(styled.primitive.angle_start);
+        let line_b = stroke_area.line_from_angle(styled.primitive.angle_end());
+
+        let line_a_iter = ThickLineIterator::new(&line_a, styled.style.stroke_width_i32());
+        let line_b_iter = ThickLineIterator::new(&line_b, styled.style.stroke_width_i32());
+
+        let iter = if !styled.style.is_transparent() {
+            DistanceIterator::new(&stroke_area)
+        } else {
+            DistanceIterator::empty()
+        };
+
+        Self {
+            iter,
+            outer_threshold,
+            outer_color: styled.style.stroke_color,
+            inner_threshold,
+            inner_color: styled.style.fill_color,
+            line_a_iter,
+            line_b_iter,
+            state: IterState::ARC,
+        }
+    }
+}
+
+impl<C> Iterator for StyledSectorIterator<C>
+where
+    C: PixelColor,
+{
+    type Item = Pixel<C>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        loop {
+            match self.state {
+                IterState::ARC => {
+                    if let Some((point, distance)) = self.iter.next() {
+                        let color = if distance < self.inner_threshold {
+                            self.inner_color
+                        } else if distance < self.outer_threshold {
+                            self.outer_color
+                        } else {
+                            None
+                        };
+
+                        if let Some(color) = color {
+                            return Some(Pixel(point, color));
+                        }
+                    } else {
+                        self.state = IterState::LINES;
+                    }
+                }
+                IterState::LINES => {
+                    if let Some(color) = self.outer_color {
+                        if let Some(point) =
+                            self.line_a_iter.next().or_else(|| self.line_b_iter.next())
+                        {
+                            break Some(Pixel(point, color));
+                        }
+                    }
+                    self.state = IterState::DONE;
+                }
+                IterState::DONE => {
+                    break None;
+                }
+            }
+        }
+    }
+}
+
+impl<'a, C: 'a> Drawable<C> for &Styled<Sector, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    fn draw<D: DrawTarget<C>>(self, display: &mut D) -> Result<(), D::Error> {
+        display.draw_sector(self)
+    }
+}
+
+impl<'a, C> IntoIterator for &'a Styled<Sector, PrimitiveStyle<C>>
+where
+    C: PixelColor,
+{
+    type Item = Pixel<C>;
+    type IntoIter = StyledSectorIterator<C>;
+
+    fn into_iter(self) -> Self::IntoIter {
+        StyledSectorIterator::new(self)
+    }
+}
+
+#[derive(Copy, Clone, PartialEq, PartialOrd, Debug)]
+struct PlaneSector {
```

Comment:
Am I right that this identical to `arc::PlaneSector`? If it is, it shouldn't be duplicated but imported.